Building a new project fails during configuration

**Describe the bug**
Configuring a new project though the Project Manager fails with the following error log:

```
CMake Error at cmake/EngineFinder.cmake:16 (file):
  file failed to open for reading (No such file or directory):

    C:/Users/Tester/O3DE/Projects/NewProject/cmake/project.json
Call Stack (most recent call first):
  CMakeLists.txt:9 (include)


CMake Error at cmake/EngineFinder.cmake:21 (message):
  Unable to read key 'engine' from 'project.json'

  Error: failed parsing json string: * Line 1, Column 1

    Syntax error: value, object or array expected.

Call Stack (most recent call first):
  CMakeLists.txt:9 (include)
```
This issue does not occur in Stabilization/2110 branch.

**Steps to reproduce**
Steps to reproduce the behavior:
1. Launch the Project Manager.
2. Create a new project.
3. Build the previously created project.

**Expected behavior**
The project is built successfully.

**Actual behavior**
The building process fails.

**Screenshots/Video**

https://user-images.githubusercontent.com/86952009/141968334-1ed00283-c6ff-4676-920c-da9951dfdf88.mp4

**Found in Branch**
Development (https://github.com/o3de/o3de/commit/3791149c6bb18d007ee375f592bdd031871f793d)

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version 10
- CPU Intel I5-9600k
- GPU RTX 2060
- Memory 16GB

--- a/Templates/DefaultProject/Template/cmake/EngineFinder.cmake
+++ b/Templates/DefaultProject/Template/cmake/EngineFinder.cmake
@@ -13,8 +13,8 @@
 include_guard()
 
 # Read the engine name from the project_json file
-file(READ ${CMAKE_CURRENT_LIST_DIR}/project.json project_json)
-set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_LIST_DIR}/project.json)
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/project.json project_json)
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/project.json)
 
 string(JSON LY_ENGINE_NAME_TO_USE ERROR_VARIABLE json_error GET ${project_json} engine)
 if(json_error)

--- a/Templates/MinimalProject/Template/cmake/EngineFinder.cmake
+++ b/Templates/MinimalProject/Template/cmake/EngineFinder.cmake
@@ -13,8 +13,8 @@
 include_guard()
 
 # Read the engine name from the project_json file
-file(READ ${CMAKE_CURRENT_LIST_DIR}/project.json project_json)
-set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_LIST_DIR}/project.json)
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/project.json project_json)
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/project.json)
 
 string(JSON LY_ENGINE_NAME_TO_USE ERROR_VARIABLE json_error GET ${project_json} engine)
 if(json_error)

